pennylane/devices/default_tensor.py

```diff
@@ -555,13 +547,15 @@ def _local_expectation(self, matrix, wires) -> float:
             Local expectation value of the matrix on the MPS.
         """
 
-        # We need to copy the MPS to avoid modifying the original state
-        qc = copy.deepcopy(self._circuitMPS)
+        # We need to copy the MPS since `local_expectation` modifies the state
+        qc = copy.deepcopy(self._quimb_mps)
```

Comment:
Is there any idea about the overhead of deep copy here?

Reply:
Unfortunately, this is a (annoying and certainly not efficient) necessary step we must follow. Otherwise, we may get incorrect results in a few cases since `local_expectation` simplifies/contracts the tensor network, and the latter is irreversibly modified. I suspect this brings a consistent amount of overhead (especially for large circuits), but at this stage, I don't know how to quantify it. I'll ask for advice from J.G. by the end of the week (it is on my TODO list)